crates/bitcoin-da/src/helpers/builders.rs

```diff
@@ -457,38 +957,235 @@ pub fn create_inscription_transactions(
+        )
+        .push_slice(PushBytesBuf::try_from(body).expect("Cannot push sequencer commitment"))
+        .push_opcode(OP_ENDIF);
+
+    // Start loop to find a 'nonce' i.e. random number that makes the reveal tx hash starting with zeros given length
+    let mut nonce: i64 = 16; // skip the first digits to avoid OP_PUSHNUM_X
+    loop {
+        if nonce % 10000 == 0 {
+            trace!(nonce, "Trying to find commit & reveal nonce");
+            if nonce > 65536 {
+                warn!("Too many iterations finding nonce");
+            }
+        }
+        let utxos = utxos.clone();
+        let recipient = recipient.clone();
+        // ownerships are moved to the loop
+        let mut reveal_script_builder = reveal_script_builder.clone();
+
+        // push first random number and body tag
+        reveal_script_builder = reveal_script_builder
+            .push_slice(nonce.to_le_bytes())
+            .push_opcode(OP_DROP);
+
+        // finalize reveal script
+        let reveal_script = reveal_script_builder.into_script();
+
+        // create spend info for tapscript
+        let taproot_spend_info = TaprootBuilder::new()
+            .add_leaf(0, reveal_script.clone())
+            .expect("Cannot add reveal script to taptree")
+            .finalize(&secp256k1, public_key)
+            .expect("Cannot finalize taptree");
+
+        // create control block for tapscript
+        let control_block = taproot_spend_info
+            .control_block(&(reveal_script.clone(), LeafVersion::TapScript))
+            .expect("Cannot create control block");
+
+        // create commit tx address
+        let commit_tx_address = Address::p2tr(
+            &secp256k1,
+            public_key,
+            taproot_spend_info.merkle_root(),
+            network,
+        );
+
+        let commit_value = (get_size(
+            &[TxIn {
+                previous_output: OutPoint {
+                    txid: Txid::from_byte_array([0; 32]),
+                    vout: 0,
+                },
+                script_sig: script::Builder::new().into_script(),
+                witness: Witness::new(),
+                sequence: Sequence::ENABLE_RBF_NO_LOCKTIME,
+            }],
+            &[TxOut {
+                script_pubkey: recipient.clone().script_pubkey(),
+                value: Amount::from_sat(reveal_value),
+            }],
+            Some(&reveal_script),
+            Some(&control_block),
+        ) as f64
+            * reveal_fee_rate
+            + reveal_value as f64)
+            .ceil() as u64;
+
+        // build commit tx
+        // we don't need leftover_utxos because they will be requested from bitcoind next call
+        let (unsigned_commit_tx, _leftover_utxos) = build_commit_transaction(
+            prev_utxo.clone(),
+            utxos,
+            commit_tx_address.clone(),
+            recipient.clone(),
+            commit_value,
+            commit_fee_rate,
+        )?;
```

Comment:
since we decided to go with wtxid for prefixes, now we have to choose utxos every time we try a nonce.

this can be improved actually. 

pls open an issue for this

Reply:
https://github.com/chainwayxyz/citrea/issues/1023